sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -545,22 +545,19 @@ def call(self):
           self._executor.executor_service.submit(self)
 
     def _should_shutdown(self):
-      """_should_shutdown checks whether pipeline is completed or not.
+      """Checks whether the pipeline is completed and should be shut down.
 
-      It will check for successful completion by checking the watermarks of all
-      transforms. If they all reached the maximum watermark it means that
-      pipeline successfully reached to completion.
+      If there is anything in the queue of tasks to do, do not shut down.
```

Comment:
<!--new_thread; commit:517fae9dcb749a59322ea40c7de0eb90ebf5efac; resolved:0-->
Are we checking whether there are pending processing time timers?